### DISCUSSION TITLE
How to make dependant fields disappear if other value selected?

### DISCUSSION THREAD

**Discussion by martonmiklos** [inventree/InvenTree#7687] [General]
I did not wanted to pollute the issue tracker with some development related help, I hope it could fit here.

I am trying to add a custom sheet size option to the label sheet printer plugin:
https://github.com/martonmiklos/InvenTree/commit/dba1a4e2145298923077e0046b03a1e9c18005ae#diff-6e924718eac2d314c838bd9e458d9a6d245a20bd486fb1e65cc30415c8eb635dR34

For revealing the fields it works great, however it does not hide them when a predefined size is selected.

Is there anything what I am missing? 

I tried to  study the machine plugin where it is utilized:
https://github.com/inventree/InvenTree/blob/master/src/backend/InvenTree/plugin/builtin/labels/inventree_machine.py#L165

However I did not found any major differences to my usage.

**Comment by martonmiklos** [comment]
The hiding partially works in the old UI, however the labels never got hidden (and visible by default):
![kép](https://github.com/user-attachments/assets/3ad2688d-954f-4f18-99fc-9a26901a1179)

![kép](https://github.com/user-attachments/assets/2401f31c-c5a0-4320-a68e-6bcb2a373bb0)


@wolflu05 would you mind to take a look on this? I am happy to fix this behavior if it is an issue (and not misuse on my side).

**Reply by wolflu05** [reply]
Hi @martonmiklos,

sorry for not replying today, I was away from the computer the whole day, so it took me some time to get to it. This is really strange behavior and definitely not expected. I tried to analyze the requests, and it seems like this is an issue with the frontend implementation. 

1. Initial OPTIONS request, with just the printing plugin selected (no child is provided for the width field):
![image](https://github.com/user-attachments/assets/5536124f-66b3-4f17-84b0-e12b36c74eff)

2. Selecting "Custom" page size (child with Integer Serializer is provided):
![image](https://github.com/user-attachments/assets/f01102d6-f2fd-4025-98b1-1922d3066e6f)

3. And again, selecting A4 (again like the first request, no child for the width field is provided):
![image](https://github.com/user-attachments/assets/e2ca3e56-28c2-46cf-b795-2cfad4f93b7c)

I would really appreciate it if you could find a fix for it and make a PR, I currently sadly have no time for fixing this.

**Comment by martonmiklos** [comment]
@wolflu05 no worries, many thanks for the quick feedback and for looking into it. 
I will try to fix it and came up with a PR.
